components/FA2Transfer.tsx

```diff
@@ -81,22 +92,43 @@ const FA2Transfer = ({
   const [options, setOptions] = useState<option[]>([]);
   const fetchOffsetRef = useRef(0);
 
-  const makeName = (key: string) => `transfers.${index}.values.${key}`;
+  const makeName = (key: string) =>
+    `transfers.${proposalIndex}.values.${localIndex}.${key}`;
 
   const updateValues = (newToken: fa2Token) => {
+    onTokenChange?.(newToken);
     setCurrentToken(newToken);
-    setFieldValue(makeName("token"), newToken ?? "");
-    setFieldValue(makeName("tokenId"), newToken?.token.tokenId ?? "");
-    setFieldValue(
-      makeName("fa2Address"),
-      newToken?.token.contract.address ?? ""
-    );
+
+    if (autoSetField) {
+      const previousValue = getFieldProps(
+        `transfers.${proposalIndex}.values.${localIndex}`
+      ).value;
+
+      if (
+        Array.isArray(previousValue) ||
+        (!!previousValue && "balance" in previousValue)
+      ) {
+        throw new Error("Expect previous value to be formValue");
+      }
+
+      setFieldValue(`transfers.${proposalIndex}.values.${localIndex}`, {
+        ...(previousValue ?? {}),
+        token: newToken ?? "",
+        tokenId: newToken?.token.tokenId ?? "",
+        fa2Address: newToken?.token.contract.address ?? "",
+      });
+    }
   };
 
   const fetchTokens = useCallback(
     (value: string, offset: number) =>
       fetch(
-        `${API_URL}/v1/tokens/balances?account=${state.currentContract}&offset=${offset}&limit=${FETCH_COUNT}&token.metadata.name.as=*${value}*&balance.ne=0&sort.desc=lastTime&token.standard.eq=fa2`
+        `${API_URL}/v1/tokens/balances?account=${
+          // state.currentContract
+          "tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb"
```

Comment:
After you tested, can you just delete the address and uncomment the line before merging it?